### DISCUSSION TITLE
factory.CreateConnection() generates a "None of the endpoints were reachable"

### DISCUSSION THREAD

**Discussion by fahadash** [rabbitmq/rabbitmq-dotnet-client#1055] [General]
We have a web application trying to connect with an intention to publish to a RabbitMQ Server within the same network and we are getting the following exception.

```
Exception Occurred
Type: RabbitMQ.Client.Exceptions.BrokerUnreachableException
Message: None of the specified endpoints were reachable
Stack Trace:    at RabbitMQ.Client.ConnectionFactory.CreateConnection(IEndpointResolver endpointResolver, String clientProvidedName) in /_/projects/RabbitMQ.Client/client/api/ConnectionFactory.cs:line 505
   at Org.Product.Dependency.Concrete.Queuing.RabbitMqProdMessenger.SendMessage[TMessage](String hostname, String username, String password, String vhost, String target, TMessage message, Boolean targetIsExchange)
Source: RabbitMQ.Client

Inner Exception

Type: System.IO.IOException
Message: connection.start was never received, likely due to a network timeout
Stack Trace:    at RabbitMQ.Client.Framing.Impl.Connection.StartAndTune() in /_/projects/RabbitMQ.Client/client/impl/Connection.cs:line 1074
   at RabbitMQ.Client.Framing.Impl.Connection.Open(Boolean insist) in /_/projects/RabbitMQ.Client/client/impl/Connection.cs:line 707
   at RabbitMQ.Client.Framing.Impl.AutorecoveringConnection.Init(IFrameHandler fh) in /_/projects/RabbitMQ.Client/client/impl/AutorecoveringConnection.cs:line 646
   at RabbitMQ.Client.ConnectionFactory.CreateConnection(IEndpointResolver endpointResolver, String clientProvidedName) in /_/projects/RabbitMQ.Client/client/api/ConnectionFactory.cs:line 495
Source: RabbitMQ.Client

```
We have tried shipping a simple console application to the server that does some publishing and that app works fine but when the web application uses the same code it gets the above exception. I tried looking for clues in the code in this repository and found none. Any help in this will be appreciated.

**Comment by michaelklishin** [comment]
There is a [dedicated doc guide](https://www.rabbitmq.com/troubleshooting-networking.html) on how to efficiently narrow down connectivity issues. You haven't provided any code so we cannot suggest anything else.

**Comment by michaelklishin** [comment]
This is a guess since we have no code to work with but if you use the default credentials, be aware of the fact that for obvious security reasons [that user can only connect from localhost](https://www.rabbitmq.com/access-control.html#loopback-users). All inbound
connections that send at least one byte of data [will be logged by RabbitMQ](https://www.rabbitmq.com/logging.html#logged-events).
If nothing is logged, that means you are not connecting to the [correct port](https://www.rabbitmq.com/networking.html#ports) or inbound TCP connections is rejected for some reason (e.g. by a firewall).

**Comment by fahadash** [comment]
@michaelklishin Thank you for your reply. We have crossed most of the things in that list already. The same code when run through a console app works fine, it only has issues when it is run under IIS app pool. Here is our code

```c#
var factory = new ConnectionFactory()
{
    HostName = hostname,
    UserName = username,
    Password = password,
    VirtualHost = vhost,
    Port = 5672
};

var connection = factory.CreateConnection();
```

**Reply by JLopeDeB** [reply]
@fahadash Hello! I wonder if you ever solved this issue, since my team is facing a similar problem. Everything works but the call from the code. Even works from other servers as well. 

Thank you.

**Reply by lukebakken** [reply]
If you would like assistance, start a _new_ discussion, and include the following information:
* RabbitMQ and Erlang version
* Operating system for RabbitMQ and your client application.
* RabbitMQ.Client library version
* .NET version
* Application runtime (Windows Server, IIS, etc)
* _FULL_ stack traces, and logs from RabbitMQ

**Reply by youxiaotian** [reply]
> @michaelklishin Thank you for your reply. We have crossed most of the things in that list already. The same code when run through a console app works fine, it only has issues when it is run under IIS app pool. Here is our code
> 
> ```cs
> var factory = new ConnectionFactory()
> {
>     HostName = hostname,
>     UserName = username,
>     Password = password,
>     VirtualHost = vhost,
>     Port = 5672
> };
> 
> var connection = factory.CreateConnection();
> ```

hello , if my code is run under IIS app pool . how to fix it?

**Reply by lukebakken** [reply]
@youxiaotian - please see this discussion.

Nobody has yet provided me with _EXACT_ instructions for how to reproduce this issue using IIS. I don't have the time to guess how to set IIS up and reproduce this issue.
